tests/integration/Dockerfile

```diff
@@ -7,6 +7,14 @@ COPY openapi.yaml .
 
 RUN pip install --no-cache-dir -r requirements.txt
 
+RUN /bin/bash -c \
+    'OUTPUT=$(head -n 1 openapi.yaml); \
+    if [[ $OUTPUT == *"swagger"* ]]; then \
+        pip install --no-cache flex; \
+    else \
+        pip install --no-cache openapi-spec-validator; \
+    fi'
+
```

Comment:
I found the post from StackOverflow here: https://stackoverflow.com/questions/29222269/is-there-a-way-to-have-a-conditional-requirements-txt-file-for-my-python-applica

> You can add certain conditional requirements after a semi-colon particularly useful for sys_platform and python_version.
>
> Examples:
>
> atomac==1.1.0; sys_platform == 'darwin'
> futures>=3.0.5; python_version < '3.0'
> futures>=3.0.5; python_version == '2.6' or python_version=='2.7'
> 
> Apparently you can also exclude particular versions of a library:
> 
> futures>=3.0,!=3.0.5
> 
> They are defined in PEP 508 and PEP 0345 (Environment Markers) but the syntax appears to follow the draft PEP 0496.


I'm not sure can we read the environment variable as the condition though. I'm good with your current method, just wonder if there is a formal way to handle these optional packages.

Reply:
Nice find, I think it might make more sense to wait until we add openAPI 3 functionality before we decide on this. Eventually, we're going to have to figure out the best way to specify the openapi version and our requirements file should be able to utilize whatever method we choose